web/packages/api/src/toEthereum_v3.ts

```diff
@@ -0,0 +1,856 @@
+                returnToSenderExecutionFeeDOT
+            ),
+        ])
+        totalFeeInNative = totalFeeInNativeRes
+        assetHubExecutionFeeNative = assetHubExecutionFeeNativeRes
+        returnToSenderExecutionFeeNative = returnToSenderExecutionFeeNativeRes
+    }
+
+    // Calculate execution cost on ethereum
+    let ethereumChain = registry.ethereumChains[registry.ethChainId.toString()]
+    let feeData = await ethereum.getFeeData()
+    let ethereumExecutionFee =
+        (feeData.gasPrice ?? 2_000_000_000n) *
+        ((tokenErcMetadata.deliveryGas ?? 100_000n) + (ethereumChain.baseDeliveryGas ?? 180_000n))
+
+    return {
+        localExecutionFeeDOT,
+        snowbridgeDeliveryFeeDOT,
+        assetHubExecutionFeeDOT,
+        bridgeHubDeliveryFeeDOT,
+        returnToSenderDeliveryFeeDOT,
+        returnToSenderExecutionFeeDOT,
+        totalFeeInDot,
+        totalFeeInNative,
+        assetHubExecutionFeeNative,
+        returnToSenderExecutionFeeNative,
+        ethereumExecutionFee,
+    }
+}
+
+export async function validateTransfer(
+    connections: {
+        sourceParachain: ApiPromise
+        assetHub: ApiPromise
+        gateway: IGateway
+        bridgeHub: ApiPromise
+    },
+    transfer: Transfer
+): Promise<ValidationResult> {
+    const { sourceParachain, gateway, bridgeHub, assetHub } = connections
+    const { registry, fee, tokenAddress, amount, beneficiaryAccount } = transfer.input
+    const {
+        sourceAccountHex,
+        sourceParaId,
+        sourceParachain: source,
+        sourceAssetMetadata,
+    } = transfer.computed
+    const { tx } = transfer
+
+    const logs: ValidationLog[] = []
+    const sourceParachainImpl = await paraImplementation(sourceParachain)
+    const nativeBalance = await sourceParachainImpl.getNativeBalance(sourceAccountHex)
+    let dotBalance: bigint | undefined = undefined
+    if (source.features.hasDotBalance) {
+        dotBalance = await sourceParachainImpl.getDotBalance(sourceAccountHex)
+    }
+    let tokenBalance: any
+    let isNativeBalance = false
+    // For DOT on AH, get it from the native balance pallet.
+    if (
+        sourceParaId == registry.assetHubParaId &&
+        transfer.computed.ahAssetMetadata.location?.parents == DOT_LOCATION.parents &&
+        transfer.computed.ahAssetMetadata.location?.interior == DOT_LOCATION.interior
+    ) {
+        tokenBalance = await sourceParachainImpl.getNativeBalance(sourceAccountHex)
+        isNativeBalance = true
+    } else {
+        tokenBalance = await sourceParachainImpl.getTokenBalance(
+            sourceAccountHex,
+            registry.ethChainId,
+            tokenAddress,
+            sourceAssetMetadata
+        )
+        isNativeBalance =
+            sourceAssetMetadata.decimals === source.info.tokenDecimals &&
+            sourceAssetMetadata.symbol == source.info.tokenSymbols
+    }
+    let nativeBalanceCheckFailed = false
+    if (isNativeBalance && fee.totalFeeInNative) {
```

Comment:
Could you explain this check, please? Perhaps with a comment? It is not clear to me why this is an error condition?

Reply:
It's inherited from the V2 API without any change. TBH, I’m not entirely sure either. I'd also assume it's not a error condition, and the variable might be better renamed to something more neutral -  like `nativeBalanceCheckRequired`. Maybe @alistair-singh can provide more context here.